### ISSUE TITLE
[java] UnusedLocalVariable false positive

### ISSUE THREAD

**Issue by kevin-wayne** [pmd/pmd#257]
**Rule Set:**
UnusedLocalVariable

**Description:**
When one of the variables is enclosed in (unnecessary) parentheses, it triggers a false positive. In the example below, the variable a gets flagged as an UnusedLocalVariable (but the variable b does not).

**Code Sample demonstrating the issue:**

```
public class Test {
    public static void main(String[] args) {
        double a = 4.0; 
        double b = 2.0; 
        double result = Math.sqrt((a) - b);
        System.out.println(result);
    }
}
```

**Running PMD 5.5.2 through: CLI

**Comment by jsotuyod** [comment]
@kevin-wayne thanks for the report. I'll be looking into this first thing tomorrow.

**Comment by jsotuyod** [comment]
The issue is confirmed. This is a symbol table analysis failure, no usages are found for variable `a`. This should affect plenty of other rules too.

I'm working on a fix.

**Comment by kevin-wayne** [comment]
Thanks for the very quick reply!

**Comment by jsotuyod** [comment]
This one was tough (a problem in the grammar itself), I have just created a PR for peer review, hopefully will be available on the next release.

**Comment by cdancy** [comment]
@jsotuyod Believe the below is similar but just posting to ensure all bases are covered as we've hit this issue recently as well.

```
Map<String, Object> helloWorld = new HashMap<>();
int index = 0;
while(iterator.hasNext()) {
    Object obj = // some code to get obj
    helloWorld.put("[:" + (index) + "]", obj);
    index++;
}
```

**Comment by jsotuyod** [comment]
@cdancy Yes, that's the same issue, `(index)` being parsed as a cast rather than as an expression between parenthesis. Thanks for submitting the sample.
